packages/core-app-api/src/app/AppManager.tsx

```diff
@@ -198,6 +202,15 @@ export class AppManager implements BackstageApp {
     return this.components;
   }
 
+  getExtensions(): ComponentExtensions {
+    return [
+      // Plugin-provided extensions
+      ...this.getPlugins().flatMap(plugin => plugin.extensions ?? []),
+      // App-provided extensions
+      ...this.extensions,
+    ];
```

Comment:
Is there any precedence when there are conflicting global extensions? An app integrator specifying conflicting extensions for the same component is probably a bug but two different plugins may provide conflicting extensions of the same component

Reply:
I wouldn't necessarily consider multiple extensions to the same component _conflicting_. Instead, it's a feature that we can have multiple registered extensions of the same component. Extending components this was is really powerful and great, but can also be a can of worms. We _might_ end up with two extension of the same component _not playing well together_. I think we'll need to resolve that on a per-case basis.

In most cases, plugins (or the app itself) will extend components in a way that won't interfere with other extensions. Well have things like:
 * adding an action button
 * setting the title of something
 * rewriting the Link "to" (_only if it matches a certain pattern_, and otherwise fallthrough)

So, when multiple extensions to, say, the Link component are registered, the effective component becomes:

```tsx
<LinkProvider to={ /* ... */ }>
  <ExtensionA>
    <ExtensionB>
      <ExtensionC>
        <LinkComponent />
      </ExtensionC>
    </ExtensionB>
  </ExtensionA>
</LinkProvider>
```

If an intermediate component doesn't have any logic for this particular link (maybe the `to` doesn't match what the extension supports), the extension will just fallthrough with `return <Component />` which renders the next extension and eventually the inner component.

Reply:
> We might end up with two extension of the same component not playing well together. I think we'll need to resolve that on a per-case basis.

Yeah that was my main concern (especially coming from different plugins who are unaware of other potential extensions) along with the ordering of plugins effecting the resulting extension outcome. 

This of course wouldn't be a concern under the opt-in model